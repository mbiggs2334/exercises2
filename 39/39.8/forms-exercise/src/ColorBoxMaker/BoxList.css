.BoxList-boxes {
    display:flex;
    justify-content: space-evenly;
    flex-wrap: wrap;
    margin: 2rem auto;
    width: 50%;
}