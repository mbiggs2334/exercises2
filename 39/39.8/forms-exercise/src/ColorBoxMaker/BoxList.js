import React, { useState } from 'react';
import Box from './Box';
import NewBoxForm from './NewBoxForm';
import {v4 as uuid } from 'uuid';
import './BoxList.css';

const BoxList = () => {

    const [boxes, setBoxes] = useState([]);

    const addBox = ({height, width, color}) => {
        setBoxes([...boxes, {height, width, color}]);
    };

    return (
        <div className={'BoxList'}>
            <NewBoxForm addBox={addBox} />
            <div className={'BoxList-boxes'}>
                {boxes.map(({height, width, color}) => 
                    <Box key={uuid()} width={width} height={height} color={color} />
                )}

            </div>
        </div>
    )
};

export default BoxList;