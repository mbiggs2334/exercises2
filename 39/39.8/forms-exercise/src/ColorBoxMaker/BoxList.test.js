import React from "react";
import { render, fireEvent } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import BoxList from './BoxList';

it('renders without crashing', () => {
    render(<BoxList />);
});

it('matches snapshot', () => {
    const { asFragment } = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
});

it('shows added boxes', () => {
    const { queryByTestId, queryByText } = render(<BoxList />);

    const heightInput = queryByTestId('height-input');
    const widthInput = queryByTestId('width-input');
    const colorInput = queryByTestId('color-input');
    const submitButton = queryByText('Add Box');

    fireEvent.change(heightInput, {target: {value: 25}})
    fireEvent.change(widthInput, {target: {value: 25}})
    fireEvent.change(colorInput, {target: {value: '#000000'}})
    fireEvent.click(submitButton);

    expect(queryByTestId('#000000')).toBeInTheDocument();

})