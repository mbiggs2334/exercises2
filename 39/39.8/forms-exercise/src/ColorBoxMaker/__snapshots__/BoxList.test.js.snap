// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form
      class="NewBoxForm-form"
    >
      <label
        for="height"
      >
        Height: 
      </label>
      <input
        data-testid="height-input"
        id="height"
        max="250"
        min="25"
        name="height"
        placeholder="Height"
        type="number"
        value=""
      />
      <label
        for="width"
      >
        Width: 
      </label>
      <input
        data-testid="width-input"
        id="width"
        max="250"
        min="25"
        name="width"
        placeholder="Width"
        type="number"
        value=""
      />
      <label
        for="color"
      >
        Color: 
      </label>
      <input
        data-testid="color-input"
        id="color"
        name="color"
        placeholder="Color"
        type="color"
        value="#000000"
      />
      <button>
        Add Box
      </button>
    </form>
    <div
      class="BoxList-boxes"
    />
  </div>
</DocumentFragment>
`;
